### PR TITLE
Refactorentrypoint

### PR DESCRIPTION
**Pull request by stramit**

### CODE REVIEW

File: Assets/BasicRenderLoopTutorial/BasicRenderLoop.cs [hunk 1]
```diff
@@ -24,31 +24,39 @@ static void CreateBasicRenderLoop()
...
-        CleanCameras(m_CamerasToRender);
-        m_CamerasToRender.Clear();
+        return new BasicRenderLoopInstance();
     }
+}
+
+public class BasicRenderLoopInstance : RenderPipeline
+{
+    
+    public override void Render(ScriptableRenderContext renderLoop, Camera[] cameras)
```
Comment (phi-lira) [inline Assets/BasicRenderLoopTutorial/BasicRenderLoop.cs]: Super minor: renderLoop name doesn't make sense here.

File: Assets/Editor/Tests/RenderloopTests/RenderloopTestFixture.cs [hunk 2]
```diff
@@ -8,25 +8,31 @@
...
+{
+    protected override IRenderPipeline InternalCreatePipeline()
+    {
+        return new BasicRenderLoopInstance();
+    }
+}
+
+public class RenderLoopTestFixtureInstance : RenderPipeline
 {
     public delegate void TestDelegate(Camera camera, CullResults cullResults, ScriptableRenderContext renderLoop);
```
Comment (phi-lira) [inline Assets/Editor/Tests/RenderloopTests/RenderloopTestFixture.cs]: Super minor: renderLoop name doesn't make sense here.

File: Assets/Editor/Tests/RenderloopTests/RenderloopTestFixture.cs [hunk 3]
```diff
@@ -8,25 +8,31 @@
...
     private static TestDelegate s_Callback;
 
     private static RenderLoopTestFixture m_Instance;
-
-    [NonSerialized]
-    readonly List<Camera> m_CamerasToRender = new List<Camera>();
-
-    public override void Render(ScriptableRenderContext renderLoop, IScriptableRenderDataStore dataStore)
+    
+    public override void Render(ScriptableRenderContext renderLoop, Camera[] cameras)
```
Comment (phi-lira) [inline Assets/Editor/Tests/RenderloopTests/RenderloopTestFixture.cs]: Super minor: renderLoop name doesn't make sense here.

File: Assets/ScriptableRenderLoop/HDRenderLoop/HDRenderLoop.cs [hunk 4]
```diff
@@ -677,15 +683,9 @@ void UpdateCommonSettings()
                 m_ShadowSettings.maxShadowDistance = m_CommonSettings.shadowMaxDistance;
             }
         }
-
-        [NonSerialized]
-        readonly List<Camera> m_CamerasToRender = new List<Camera>();
-
-        public override void Render(ScriptableRenderContext renderLoop, IScriptableRenderDataStore dataStore)
+        
+        public void Render(ScriptableRenderContext renderLoop, IEnumerable<Camera> cameras)
```
Comment (phi-lira) [inline Assets/ScriptableRenderLoop/HDRenderLoop/HDRenderLoop.cs]: Super minor: renderLoop name doesn't make sense here.

File: Assets/ScriptableRenderLoop/fptl/FptlLighting.cs [hunk 5]
```diff
@@ -1009,16 +1012,10 @@ int GenerateSourceLightBuffers(Camera camera, CullResults inputs)
 
             return numLightsOut + numProbesOut;
         }
-
-        [NonSerialized]
-        readonly List<Camera> m_CamerasToRender = new List<Camera>();
-
-        public override void Render(ScriptableRenderContext renderLoop, IScriptableRenderDataStore dataStore)
+        
+        public void Render(ScriptableRenderContext renderLoop, IEnumerable<Camera> cameras)
```
Comment (phi-lira) [inline Assets/ScriptableRenderLoop/fptl/FptlLighting.cs]: Super minor: renderLoop name doesn't make sense here.

File: Assets/ScriptableRenderLoop/HDRenderLoop/HDRenderLoop.cs [hunk 6]
```diff
@@ -6,9 +6,36 @@
...
 {
+    public class HDRenderLoopInstance : RenderPipeline
+    {
+        private readonly HDRenderLoop m_Owner;
+
+        public HDRenderLoopInstance(HDRenderLoop owner)
+        {
+            m_Owner = owner;
+
+            if (m_Owner != null)
```
Comment (vlad-andreev) [inline Assets/ScriptableRenderLoop/HDRenderLoop/HDRenderLoop.cs]: is there a valid situation where the owner is null?  if not, it should throw an exception here

File: Assets/ScriptableRenderLoop/HDRenderLoop/HDRenderLoop.cs [hunk 7]
```diff
@@ -6,9 +6,36 @@
...
+                m_Owner.Build();
+        }
+        
+        public override void Dispose()
+        {
+            base.Dispose();
+            if (m_Owner != null)
```
Comment (vlad-andreev) [inline Assets/ScriptableRenderLoop/HDRenderLoop/HDRenderLoop.cs]: same

**Pull request merged**